crates/rattler_installs_packages/src/artifacts/sdist.rs

```diff
@@ -883,4 +883,49 @@ mod tests {
 
         assert_debug_snapshot!(wheel_metadata.1);
     }
+
+    #[tokio::test(flavor = "multi_thread")]
+    pub async fn get_whl_for_local_stree_rich() {
+        let path = Path::new(env!("CARGO_MANIFEST_DIR"))
+            .join("../../test-data/stree/dev_folder_with_rich");
+
+        let url = Url::from_file_path(path.canonicalize().unwrap()).unwrap();
+
+        let package_db = get_package_db();
+        let env_markers = Arc::new(Pep508EnvMakers::from_env().await.unwrap().0);
+        let wheel_builder = WheelBuilder::new(
+            package_db.0.clone(),
+            env_markers,
+            None,
+            ResolveOptions::default(),
+            HashMap::default(),
+        )
+        .unwrap();
+
+        let norm_name = PackageName::from_str("rich").unwrap();
+        let stree_file_name = STreeFilename {
+            distribution: norm_name,
+            version: Version::from_str("0.0.0").unwrap(),
+            url: url.clone(),
+        };
+
+        let artifact_info = ArtifactInfo {
+            filename: ArtifactName::STree(stree_file_name),
+            url: url,
+            hashes: None,
+            requires_python: None,
+            dist_info_metadata: DistInfoMetadata::default(),
+            yanked: Yanked::default(),
+        };
+
+        let whl = package_db
+            .0
+            .get_wheel(&artifact_info, Some(&wheel_builder))
+            .await
+            .unwrap();
+
+        let whl_metadata = whl.metadata().unwrap();
```

Comment:
Would be good to actually install into a temporary `venv` just to make sure it works.